app/src/main/java/net/squanchy/notification/NotificationCreator.java

```diff
@@ -0,0 +1,159 @@
+                .setContentIntent(createPendingIntentForMultipleEvents())
+                .setContentTitle(createSummaryTitle(events.size()))
+                .setGroup(GROUP_KEY_NOTIFY_SESSION)
+                .setGroupSummary(true)
+                .setLocalOnly(true);
+
+        NotificationCompat.InboxStyle richNotification = createInboxStyleRichNotification(summaryBuilder, events);
+
+        return richNotification.build();
+    }
+
+    private NotificationCompat.Builder createDefaultBuilder(int talksCount) {
+        Resources resources = context.getResources();
+
+        NotificationCompat.WearableExtender extender = new NotificationCompat.WearableExtender();
+        extender.setBackground(BitmapFactory.decodeResource(resources, R.drawable.notification_background));
+        // TODO: update notification background
+
+        return new NotificationCompat.Builder(context)
+                .setTicker(
+                        context.getResources().getQuantityString(
+                                R.plurals.event_notification_ticker,
+                                talksCount,
+                                talksCount
+                        )
+                )
+                .setDefaults(Notification.DEFAULT_SOUND | Notification.DEFAULT_VIBRATE)
+                .setLights(
+                        resources.getColor(R.color.notification_led_color, context.getTheme()),
+                        NOTIFICATION_LED_ON_MS,
+                        NOTIFICATION_LED_OFF_MS
+                )
+                .setSmallIcon(R.drawable.ic_stat_notification)
+                .setPriority(Notification.PRIORITY_MAX)
+                .setAutoCancel(true)
+                .extend(extender);
+    }
+
+    private PendingIntent createPendingIntentForSingleEvent(int day, long eventId) {
+        TaskStackBuilder taskBuilder = createBaseTaskStackBuilder();
+        Intent eventDetailIntent = EventDetailsActivity.createIntent(context, day, eventId);
+        taskBuilder.addNextIntent(eventDetailIntent);
+
+        return taskBuilder.getPendingIntent(0, PendingIntent.FLAG_CANCEL_CURRENT);
+    }
+
+    private PendingIntent createPendingIntentForMultipleEvents() {
+        TaskStackBuilder taskBuilder = createBaseTaskStackBuilder();
+        return taskBuilder.getPendingIntent(0, PendingIntent.FLAG_CANCEL_CURRENT);
+    }
+
+    private TaskStackBuilder createBaseTaskStackBuilder() {
+        Intent baseIntent = new Intent(context, ScheduleActivity.class);
+        baseIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        return TaskStackBuilder.create(context)
+                .addNextIntent(baseIntent);
+    }
+
+    private NotificationCompat.BigTextStyle createBigTextRichNotification(NotificationCompat.Builder notificationBuilder, Event event) {
+        StringBuilder bigTextBuilder = new StringBuilder()
+                .append(getDisplayedSpeakers(event));
+        if (event.place() != null) {
```

Comment:
Not sure. I think we need to redefine our domain models (the ones you're using are viewmodels), maybe we can sync up with @fourlastor and @tiwiz on this?